ruby/spec/timeline_spec.rb

```diff
@@ -64,6 +64,53 @@
   end
 
   describe 'validations' do
-    # Requires TimelineEvent table
+    before(:all) do
+      SpecTables.create_versioned_addresses_table
+      SpecTables.create_timeline_events_table
+    end
+    after(:all) do
+      SpecTables.drop_table('timeline_events')
+      SpecTables.drop_table('versioned_addresses')
+    end
+
+    subject do
+      described_class.create(
+        uuid: timeline_uuid,
+        timeline_events: timeline_events,
+        transaction_start: DateTime.new(2019, 1, 1),
+        transaction_stop: DateTime.new(2019, 2, 1),
+      )
+    end
+    let(:timeline_uuid) { uuid }
+    let(:timeline_events) { versions.map { |v| Bitemporal::TimelineEvent.new(version: v) } }
+    let(:versions) { [version_1, version_2] }
+
+    let(:versioned_address_class) do
+      Class.new(ActiveRecord::Base) do
+        self.table_name = 'versioned_addresses'
+        include Bitemporal::Versioned
+      end
+    end
+    let(:version_1) do
+      versioned_address_class.create!(
+        uuid: version_1_uuid,
+        effective_start: DateTime.new(2019, 1, 1),
+        effective_stop: DateTime.new(2019, 2, 1),
+        street_1: '1 Infinity Loop',
+      )
+    end
+    let(:version_2) do
+      versioned_address_class.create!(
+        uuid: version_2_uuid,
+        effective_start: DateTime.new(2019, 2, 1),
+        effective_stop: DateTime.new(2019, 3, 1),
+        street_1: '123 Infinity Loop',
+      )
+    end
+
+    let(:version_1_uuid) { uuid }
+    let(:version_2_uuid) { uuid }
+
+    it { is_expected.to be_valid }
```

Comment:
All this test setup for just this one little assertion? Eeek. I think we're going to need to reconsider.